File: src/main/java/co/gatedaccess/web/config/WebConfig.java
package co.gatedaccess.web.config;

import co.gatedaccess.web.SecureInterceptor;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.servlet.config.annotation.InterceptorRegistry;
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;

@Configuration
public class WebConfig implements WebMvcConfigurer {

    @Override
    public void addInterceptors(InterceptorRegistry registry) {
        registry.addInterceptor(new SecureInterceptor()).addPathPatterns("/secure/**");
    }
}

File: src/main/kotlin/co/gatedaccess/web/config/WebConfig.kt
package co.gatedaccess.web.config

import co.gatedaccess.web.http.interceptor.SecureInterceptor
import org.springframework.context.annotation.Configuration
import org.springframework.web.servlet.config.annotation.EnableWebMvc
import org.springframework.web.servlet.config.annotation.InterceptorRegistry
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer

@Configuration
@EnableWebMvc
open class WebConfig : WebMvcConfigurer {
    override fun addInterceptors(registry: InterceptorRegistry) {
        registry.addInterceptor(SecureInterceptor()).addPathPatterns("/secure/**")
    }
}
